### PR TITLE
Add -U to bsmap to generate unsorted bam

### PR DESCRIPTION
**Pull request by Dangertrip**

### CODE REVIEW

File: src/bsmap/main.cpp [hunk 1]
```diff
@@ -206,6 +206,7 @@ cerr<<"Usage:	bsmap [options]\n"
 		<<"       -o  <str>   output alignment file, BSP/SAM/BAM format, if omitted, the output will be written to STDOUT in SAM format.\n"
 		<<"\n  Options for alignment:\n"
 		<<"       -s  <int>   seed size, default=16(WGBS mode), 12(RRBS mode). min=8, max=16.\n"
+    <<"       -U  <int>   leave the output bam unsorted(reads will follow the order of the input fastq files), default="<<param.unsort<<"\n\n"
```
Comment (lijinbio) [inline src/bsmap/main.cpp]: No need to set it as an `int`. Please check `-u` or `-R`.

Comment (Dangertrip) [reply src/bsmap/main.cpp]: Added another commit and changed all int to bit32_t

File: src/bsmap/main.cpp [hunk 2]
```diff
@@ -206,6 +206,7 @@ cerr<<"Usage:	bsmap [options]\n"
 		<<"       -o  <str>   output alignment file, BSP/SAM/BAM format, if omitted, the output will be written to STDOUT in SAM format.\n"
 		<<"\n  Options for alignment:\n"
 		<<"       -s  <int>   seed size, default=16(WGBS mode), 12(RRBS mode). min=8, max=16.\n"
+        <<"       -U          leave the output bam unsorted(reads will follow the order of the input fastq files), default="<<param.unsort<<"\n\n"
```
Comment (lijinbio) [inline src/bsmap/main.cpp]: It is more about reporting as unsorted BAM, not alignment parameters. Move it to the reporting section. Ensure no extra newline.

Comment (Dangertrip) [reply src/bsmap/main.cpp]: Solved

File: src/bsmap/main.cpp [hunk 3]
```diff
@@ -250,6 +250,7 @@ cerr<<"Usage:	bsmap [options]\n"
 		<<"       -H          do not print header information in SAM format output\n"
 		<<"       -V  [0,1,2] verbose level: 0=no message displayed (quiet mode); \n"
 		<<"                   1=major message (default); 2=detailed message.\n" 
+        <<"       -U          leave the output bam unsorted(reads will follow the order of the input fastq files), default="<<param.unsort<<"\n\n"
```
Comment (lijinbio) [inline src/bsmap/main.cpp]: Change to `default=off`.  Do not add an extra new line.

**Pull request closed without merging**